chainerx_cc/chainerx/cuda/memory_pool_test.cc

```diff
@@ -12,14 +12,94 @@ namespace cuda_internal {
 
 class MemoryPoolTest {
 public:
-    static const std::vector<std::vector<void*>>& GetFreeBins(const MemoryPool& pool) { return pool.free_bins_; }
+    static const FreeBinsMap& GetFreeBins(const MemoryPool& pool) { return pool.free_bins_; }
     static const Allocator* GetAllocator(const MemoryPool& pool) { return pool.allocator_.get(); }
 };
 
 }  // namespace cuda_internal
 
 namespace {
 
+void* AddOffset(void* ptr, size_t offset) {
+    return reinterpret_cast<void*>(reinterpret_cast<intptr_t>(ptr) + offset);  // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
+}
+
+TEST(ChunkTest, Split) {
```

Comment:
I'm having hard time understanding these unit tests.
Could you add some comments describing the objectives?